resources/assets/js/device.js

```diff
@@ -0,0 +1,26 @@
+$(document).ready(function () {
+  function validateBeforeSubmit(event) {
+    event.preventDefault();
+    $(this).find('#device-name-input-message').empty();
+    axios.get('/api/devices')
+    .then((response) => {
+      var deviceName = $(this).find('#device-name-input').val();
+      var devices = response.data.payload.discoveredAppliances;
+
+      for (let i=0; i<devices.length; ++i) {
+        if (deviceName == devices[i].friendlyName) {
+          $(this).find('#device-name-input-message').text("Device with '" + deviceName + "' already exists!");
+          return false;
+        }
+      }
+
+      this.submit();
+    })
+    .catch(function (error) {
+      console.log(error);
```

Comment:
It's not possible to write log back to server side from client unless you create an API endpoint or something that continuously receive log from any client. If it's not a good idea to show in client side, it can be removed.

Reply:
That's fine, just wondering.  Let's leave it like this for now, it can be iterated on later if necessary.